Fixup rpimem to cleanly unload and reload, and make checkpatch clean.

6by9 opened the pull request:
Issue noted when JC was trying to unload and reload an overlay for the V4L2 HEVC codec.
Made checkpatch clean (other than DT compatible string) whilst I was at it.

6by9 committed:
driver: char: rpivid: Destroy the legacy device on remove

The legacy name support created a new device that was never destroyed.
If the driver was unloaded and reloaded, it failed due to the
device already existing.

Fixes: "75f1d14 driver: char: rpivid - also support legacy name"
Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
driver: char: rpivid: Add error handling to the legacy device load

The return value from device_create for the legacy device was never
checked or handled. Add the required error handling.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
driver: char: rpivid: Fix coding style whitespace issues.

Makes checkpatch happier.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
driver: char: rpimem: Add SPDX licence header.

Stops checkpatch complaining.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

pelwell commented on drivers/char/broadcom/rpivid-mem.c:
```diff
@@ -197,22 +197,29 @@ static int rpivid_mem_probe(struct platform_device *pdev)
 		kfree(oldname);
+
+		ptr_err = old_mem_dev;
+		if (IS_ERR(old_mem_dev))
```
IS_ERR(ptr_err) would seem more logical here. Or just drop the old_mem_dev and assign directly to ptr_err.

6by9 replied:
I'd copied and pasted from the other device_create failure case. Having ptr_err seems redundant anyway, and the error case may as well do ```err = PTR_ERR(rpivid_mem_dev)``` or ```err = PTR_ERR(rpivid_mem_dev)```.

Actually the handling of ```dev``` as being ```&pdev->dev```, and is then assigned to ```priv-dev``` leaves ```dev``` unused through the rest of the function. I'll do a few more lines of cleanup.

pelwell commented on drivers/char/broadcom/rpivid-mem.c:
```diff
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
The SPDX declaration doesn't appear to match the licence text, which looks like a dual 3-clause BSD and GPL2.

6by9 replied:
Oops, missed that!

6by9 committed:
driver: char: rpivid: Clean up error handling use of ERR_PTR/IS_ERR

The driver used an unnecessary intermediate void* variable so it
only called ERR_PTR once to convert to the error value.

Switch to converting as the error arises to remove these intermediate
variables.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
driver: char: rpivid: Fix access to freed memory

The error path during probe frees the private memory block, and
then promptly dereferences it to log an error message.

Use the base device instead of the pointer to it in the private
structure.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 commented:
Updated. Reworked the conversion from PTR to ERR overall. 
Added an extra patch to remove a deref of ```priv``` when we've just freed it too!
